tests/unit/insertTestResult.unitTest.ts

```diff
@@ -3563,5 +3562,385 @@ describe('insertTestResult', () => {
+                linkedTestCode: null,
+                defaultTestCode: 'qjt1',
+                testTypeClassification: 'IVA With Certificate',
+              }),
+            getBySystemNumber: (systemNumber: any) => Promise.resolve([]),
+          }));
+
+          testResultsService = new TestResultsService(new MockTestResultsDAO());
+
+          expect.assertions(2);
+          return testResultsService
+            .insertTestResult(testResult)
+            .then((insertedTestResult: any) => {
+              expect(insertedTestResult[0].testTypes[0].testTypeId).toBe('125');
+              expect(insertedTestResult[0].testTypes[0].certificateNumber).toBe(
+                'W01A00209',
+              );
+            });
+        });
+      },
+    );
+
+    context(
+      'when creating a test record for an IVA test with a certificate number',
+      () => {
+        it('should create the record succesfully and the certificate number should match the one provided', () => {
+          const testResult = cloneDeep(testResultsPostMock[13]);
+          testResult.testTypes[0].testTypeId = '125';
+          testResult.testTypes[0].certificateNumber = '12345';
+          testResult.testTypes[0].requiredStandards?.push({
+            sectionNumber: '01',
+            sectionDescription: 'Noise',
+            rsNumber: 1,
+            requiredStandard: 'The exhaust must be securely mounted.',
+            refCalculation: '1.1',
+            additionalInfo: true,
+            inspectionTypes: [],
+            prs: false,
+            additionalNotes: '',
+          });
+          MockTestResultsDAO = jest.fn().mockImplementation(() => ({
+            createSingle: () =>
+              Promise.resolve({
+                Attributes: Array.of(testResult),
+              }),
+            createTestNumber: () =>
+              Promise.resolve({
+                testNumber: 'W01A00209',
+                id: 'W01',
+                certLetter: 'A',
+                sequenceNumber: '002',
+              }),
+            getTestCodesAndClassificationFromTestTypes: () =>
+              Promise.resolve({
+                linkedTestCode: null,
+                defaultTestCode: 'qjt1',
+                testTypeClassification: 'IVA With Certificate',
+              }),
+            getBySystemNumber: (systemNumber: any) => Promise.resolve([]),
+          }));
+
+          testResultsService = new TestResultsService(new MockTestResultsDAO());
+
+          expect.assertions(2);
+          return testResultsService
+            .insertTestResult(testResult)
+            .then((insertedTestResult: any) => {
+              expect(insertedTestResult[0].testTypes[0].testTypeId).toBe('125');
+              expect(insertedTestResult[0].testTypes[0].certificateNumber).toBe(
+                '12345',
+              );
+            });
+        });
+      },
+    );
+
+    context(
+      'when creating a test record for an IVA test with no certificate number',
```

Comment:
As the first test you did had the context 'when creating a test record for an IVA test with no certificate number', would it be worth renaming this to 'when creating a test record for an IVA test with null as certificate number' or something similar, to differentiate it from the first one?

Reply:
Good spot, risk of having the test data in a separate file. will remove this test.